extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js

```diff
@@ -0,0 +1,375 @@
+						this.showFirst();
+
+						this.bucky.timer.stop('markAllAsReadRequest');
+					})
+				});
+			},
+
+			markAllAsReadAllWikis: function (e) {
+				this.markAllAsReadRequest('FORCE');
+				return false;
+			},
+
+			showFirst: function () {
+				var wikiEl = this.$wallNotifications.find('.notifications-for-wiki').first(),
+					firstWikiId = parseInt(wikiEl.data('wiki-id'), 10);
+
+				if ( !isNaN(firstWikiId) ) {
+					wikiEl.addClass('show');
+					this.wikiShown = {};
+					this.wikiShown[firstWikiId] = true;
+				}
+			},
+
+			updateCountsHtml: function (data) {
+				var self = this,
+					element;
+
+				this.$wallNotifications.html(data.html);
+				this.unreadCount = data.count;
+
+				if (data.count > 0) {
+					this.$notificationsCount.html(data.count).parent('.bubbles').addClass('show');
+					this.fetchForCurrentWiki();
+					this.$wallNotifications.addClass('show');
+				} else {
+					this.$notificationsCount.empty().parent('.bubbles').removeClass('show');
+				}
+
+				this.$wallNotifications.find('.notifications-wiki-header').click(this.wikiClick.bind(this));
+
+				this.setNotificationsHeight();
+			},
+
+			wikiClick: function (e) {
+				var wikiEl = $(e.target).closest('.notifications-for-wiki'),
+					wikiId = parseInt(wikiEl.data('wiki-id'), 10);
+
+				if (wikiEl.hasClass('show')) {
+					wikiEl.removeClass('show');
+					delete this.wikiShown[wikiId];
+					this.setNotificationsHeight();
+				} else {
+					wikiEl.addClass('show');
+					this.wikiShown[wikiId] = true;
+					this.updateWiki(wikiId);
+				}
+
+				return false;
+			},
+
+			updateWiki: function (wikiId) {
+				if (!(wikiId in this.notificationsCache)) {
+					// nothing in cache, just fetch
+					this.updateWikiFetch(wikiId);
+					return;
+				}
+				var data = this.notificationsCache[wikiId],
+					wikiCount = parseInt(this.$wallNotifications.find(
+						'li[data-wiki-id=' + wikiId + ']'
+					).data('unread-count'), 10);
+				// no change in notifications, update from cache
+				if (wikiCount === data['unread']) {
+					this.updateWikiHtml(wikiId, data);
+
+					// different number of unread, fetch from server
+					// in the meantime, show old data
+				} else {
+					this.updateWikiHtml(wikiId, data);
+					this.updateWikiFetch(wikiId);
+				}
+			},
+
+			updateWikiFetch: function (wikiId) {
+				var isCrossWiki = (wikiId === this.cityId) ? '0' : '1',
+					data = {
+						wikiId: wikiId,
+						isCrossWiki: isCrossWiki
+					};
+
+				$.extend(data, this.getUrlParams());
+
+				nirvana.sendRequest({
+					controller: 'WallNotificationsExternalController',
+					method: 'getUpdateWiki',
+					type: 'GET',
+					data: data,
+					callback: this.proxy(function (data) {
+						if (data.status !== true || data.html === '') {
+							return;
+						}
+						this.updateWikiHtml(wikiId, data);
+						this.notificationsCache[wikiId] = data;
+					}),
+					onErrorCallback: this.proxy(function (jqXHR, textStatus, errorThrown) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameters "textStatus, errorThrown". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)